core/src/main/java/org/elasticsearch/index/percolator/PercolatorQueriesRegistry.java

```diff
@@ -187,7 +187,7 @@ Query parsePercolatorDocument(String id, BytesReference source) {
     private Query parseQuery(String type, XContentParser parser) {
         String[] previousTypes = null;
         if (type != null) {
-            QueryShardContext.setTypesWithPrevious(new String[]{type});
+            QueryShardContext.setTypesWithPrevious(type);
```

Comment:
this is a different issue, but while we change this line, I think we should also set the `previousTypes` variable :)